File: JobPortalFrontEnd/job-portal-ui/src/types/Job.ts
export interface Job {
    id: number;
    title: string;
    description: string;
    location?: string;
    employmentType?: string;
    experienceLevel?: string;
    minSalary?: number;
    maxSalary?: number;
    status?: string;
    deadline?: string;
    companyName?: string;
    createdAt?: string;
}

export interface JobRequest {
    title: string;
    description: string;
    location?: string;
    refEmploymentType?: number;
    experienceLevel?: string;
    minSalary?: number;
    maxSalary?: number;
    deadline?: string;
}

File: JobPortalAPI/Models/Common/Job.cs
namespace JobPortalAPI.Models.Common
{
    public class Job
    {
        public long Id { get; set; }
        public long CompanyId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string? Location { get; set; }
        public long? RefEmploymentType { get; set; }
        public string? ExperienceLevel { get; set; }
        public decimal? MinSalary { get; set; } 
        public decimal? MaxSalary { get; set; } 
        public string? Status { get; set; }
        public DateOnly? Deadline { get; set; }
        public long CreatedBy { get; set; }
        public long UpdatedBy { get; set; }
    }
}
